tests/acceptance/new-test.js

```diff
@@ -646,6 +650,65 @@ describe('Acceptance: ember new', function () {
 
       checkFileWithEmberCLIVersionReplacement(fixturePath, 'tests/dummy/config/ember-cli-update.json');
     });
+
+    it('app + typescript', async function () {
+      // This is a very slow test, as the blueprint installs ember-cli-typescript, which requires installing all dependencies,
+      // regardless of --skip-npm
+      this.timeout(600000);
+
+      // we have to use yarn here, as npm fails on unresolvable peer dependencies, see https://github.com/emberjs/ember-test-helpers/issues/1236
+      await ember(['new', 'foo', '--typescript', '--skip-npm', '--skip-bower', '--skip-git', '--yarn']);
+
+      let fixturePath;
+      if (isExperimentEnabled('EMBROIDER')) {
+        fixturePath = 'app/typescript-embroider';
+      } else {
+        fixturePath = 'app/typescript';
+      }
+
+      // check fixtures
+      ['.ember-cli'].forEach((filePath) => {
+        checkFile(filePath, path.join(__dirname, '../fixtures', fixturePath, filePath));
+      });
+      checkFileWithEmberCLIVersionReplacement(fixturePath, 'config/ember-cli-update.json');
+      checkEslintConfig(fixturePath);
+
+      // smoke test for the existence of essential TypeScript features...
+      // we are deliberately *not* comparing the package.json against a fixture here, as we have delegated essential
+      // TS setup to ember-cli-typescript's own blueprint. Instead, we are relying on its own test coverage, otherwise
+      // we would get very brittle tests
+      let pkgJson = fs.readJsonSync('package.json');
+      expect(pkgJson.scripts['lint:ts']).to.equal('tsc --noEmit');
+      expect(pkgJson.devDependencies['ember-cli-typescript']).to.exist;
+      expect(pkgJson.devDependencies['typescript']).to.exist;
+      expect(Object.keys(pkgJson.devDependencies).some((pkgName) => pkgName.match(/^@types/))).to.be.true;
+
+      expect(file('tsconfig.json')).to.exist;
+    });
+
+    it('addon + typescript', async function () {
+      this.timeout(600000);
+
+      await ember(['addon', 'foo', '--typescript', '--skip-npm', '--skip-bower', '--skip-git', '--yarn']);
+
+      let fixturePath = 'addon/typescript';
+
+      // check fixtures
+      ['.ember-cli'].forEach((filePath) => {
+        checkFile(filePath, path.join(__dirname, '../fixtures', fixturePath, filePath));
+      });
+      checkFileWithEmberCLIVersionReplacement(fixturePath, 'tests/dummy/config/ember-cli-update.json');
+      checkEslintConfig(fixturePath);
+
+      // smoke test for the existence of essential TypeScript features... (see app test)
+      let pkgJson = fs.readJsonSync('package.json');
+      expect(pkgJson.scripts['lint:ts']).to.equal('tsc --noEmit');
+      expect(pkgJson.devDependencies['ember-cli-typescript']).to.exist;
```

Comment:
```suggestion
      expect(pkgJson.dependencies['ember-cli-typescript']).to.exist;
```

`ember-cli-typescript` needs to end up in `dependencies` for addons:

```
WARNING: `ember-cli-typescript` should be included in your `dependencies`, not `devDependencies`
```

Reply:
Great catch! Fixed.